engine/api/ab_test/routes.js

```diff
@@ -31,15 +31,16 @@ module.exports = (appConfig, routeA, routeB, newrelic) => {
 
   router.use((req, res, next) => {
     if (req.url !== '/') return next()
-   
+
     if (isTestRunning) {
-      currentRoute = currentRoute === routeA ? routeB : routeA
+      currentRoute = Math.random() < 0.5 ? routeA : routeB
     }
     else {
       logger.info('Test ended, serving a')
     }
 
     logger.info(`serving ${currentRoute}`)
+    newrelic.recordCustomEvent('pageView', { page_version: currentRoute === routeA ? 'a' : 'b' })
```

Comment:
Can you add a check here to make sure `newrelic` is defined? That way it won't error if it's not instrumented.
```suggestion
    if (newrelic) {
        newrelic.recordCustomEvent('pageView', { page_version: currentRoute === routeA ? 'a' : 'b' })
    }
```

Reply:
@aswanson-nr done!